packages/core/src/rules/common/tags-duplicated-names.ts

```diff
@@ -0,0 +1,36 @@
+import { getTagName } from '../utils.js';
+
+import type { Oas3Definition, Oas3_1Definition } from '../../typings/openapi.js';
+import type { Oas2Definition } from '../../typings/swagger.js';
+import type { Async2Rule, Async3Rule, Oas2Rule, Oas3Rule } from '../../visitors.js';
+import type { UserContext } from '../../walk.js';
+
+export const TagsDuplicatedNames: Oas3Rule | Oas2Rule | Async2Rule | Async3Rule = ({
+  ignoreCase = false,
+}) => {
+  const tagNames = new Set<string>();
+
+  return {
+    Tag: {
+      leave(
+        tag: { name: string },
+        ctx: UserContext,
+        root: Oas2Definition | Oas3Definition | Oas3_1Definition
+      ) {
+        if (!root.tags) return;
+
+        const tagName = getTagName(tag, ignoreCase);
+        const index = root.tags.findIndex((t) => t === tag);
+
+        if (tagNames.has(tagName)) {
+          ctx.report({
+            message: `Duplicate tag name found: '${tag.name}'`,
+            location: ctx.location.child(['tags', index]),
```

Comment:
You don't need to specify the child. I believe you already have the correct location in `ctx` since you are inside the `Tag` visitor. You may use it like here: https://github.com/Redocly/redocly-cli/blob/main/packages/core/src/rules/oas3/array-parameter-serialization.ts#L29.